README.md

```diff
@@ -1,109 +1,56 @@
-#### IntelliJ
-Module Settings > Dependencies > click the + > JARs or directories > Select your JAR file
-#### Eclipse
-Project Properties > Java Build Path > Add the jar file
+## Quick Example
+```java
+final DiscordClient client = new ClientBuilder("token").build();
 
-## So, how do I use this?
-### Tutorials/Resources
-* TODO
+client.getEventDispatcher().on(ReadyEvent.class)
+        .subscribe(ready -> System.out.println("Logged in as " + ready.getSelf().getUsername()));
 
-### Starting with the API
-TODO
+client.getEventDispatcher().on(MessageCreateEvent.class)
+        .map(MessageCreateEvent::getMessage)
+        .filter(msg -> msg.getContent().map(content -> content.equals("!ping")).orElse(false))
+        .flatMap(Message::getChannel)
+        .flatMap(channel -> channel.createMessage(spec -> spec.setContent("Pong!")))
+        .subscribe();
 
-### More examples
-TODO
-
-## Projects using Discord4J
-TODO
+client.login().block();
+```
 
-## Deprecation policy
-Due to the nature of the Discord API, in general any deprecations found in the API should not be expected to last past the current
-version. Meaning that if a method is deprecated on version 2.1.0, do not assume the method will be available in version 2.2.0.
+## Modules
+Discord4J is highly oriented towards customizability and fine-grained control. To achieve this, the project is divided into several "modules" which can be used separately depending on your use case.
 
-In some extraordinary cases, we may be forced to remove APIs without going through a deprecation cycle.
+### [Core](./core/README.md)
+The `core` module combines the other modules to form high-level abstractions for the entire Discord Bot API. This is the module most users will want when making bots.
 
-## Development
-The Discord API is still in development. Functions may break at any time.  
-In such an event, please contact me or submit a pull request.
+### [Rest](./rest/README.md)
+The `rest` module provides a low-level HTTP client specifically for Discord which properly handles Discord's [ratelimiting system](https://discordapp.com/developers/docs/topics/rate-limits).
 
-## Pull requests
-No one is perfect at programming and I am no exception. If you see something that can be improved, please read the [contributing guildelines](https://github.com/Discord4J/Discord4J/blob/v3/.github/CONTRIBUTING.md) and feel free to submit a pull request! 
+### [Gateway](./gateway/README.md)
+The `gateway` module provides a low-level WebSocket client for interacting with the [Discord Gateway](https://discordapp.com/developers/docs/topics/gateway).
 
-## Other info
-More information can be found in the official [javadocs](http://discord4j.com/docs.html). 
+### [Store](./store/README.md)
+The `store` module provides a platform for the efficient caching of Discord gateway data.
 
```

Comment:
Mfw no common module

Reply:
I'm still not sure this is really a module for user consumption. I can't imagine a way in which it is useful to anyone.

Reply:
Probably not, but even a description saying that is fine